### DISCUSSION TITLE
[Docs] Document effective abstraction patterns

### DISCUSSION THREAD

**Discussion by kpenfound** [dagger/dagger#12092] [maintainers]
### What is the issue?

We should provide documentation/guidance for effective methods of abstracting + reusing (within a repo or org):
- pipeline components. wrappers for building parts of a container that are reused across multiple parts of a pipeline
- subpipelines. Reusing entire subpipelines across multiple pipelines like a github action
- entire pipelines. Reuse common pipelines across multiple repos without reimplementing

**Comment by mircubed** [comment]
@vikram-dagger a user requested this because they are trying to understand how to fit Dagger into their use case properly. I think this could help many other users understand best practices too.
